src/language/language_type.hbs.rs

```diff
@@ -396,29 +418,53 @@ impl LanguageType {
         }
 
         let extension = fsutils::get_extension(&entry);
-        let filetype = extension.as_ref()
-            .map(|s| &**s)
-            .or_else(|| get_filetype_from_shebang(&entry));
-
-        if let Some(extension) = filetype {
-            /*
-            if let Some(ref languages) = config.languages {
-                for (language, lang_config) in languages {
-                    if lang_config.extensions.iter().any(|e| e == extension) {
-                        return Some(*language)
-                    }
-                }
-            }
-            */
+        match extension {
+            Some(extension) => LanguageType::from_file_extension(extension.as_str()),
+            None => LanguageType::from_shebang(&entry),
+        }
+    }
 
-            match extension {
+    /// Get language from a file extension.
+    ///
+    /// ```no_run
+    /// use tokei::LanguageType;
+    ///
+    /// let rust = LanguageType::from_file_extension("rs");
+    ///
+    /// assert_eq!(rust, Some(LanguageType::Rust));
+    /// ```
+    pub fn from_file_extension(extension: &str) -> Option<Self> {
+        match extension {
+            {{~#each languages}}
+                {{~#if this.extensions}}
+                    {{~#each this.extensions}}
+                        | "{{~this}}"
+                    {{~/each}}
+                        => Some({{~@key}}),
+                {{~/if}}
+            {{~/each}}
+            extension => {
+                warn!("Unknown extension: {}", extension);
+                None
+            },
+        }
+    }
+
+    /// Get language from a shebang. May open and read the file.
+    ///
+    /// ```no_run
+    /// use tokei::LanguageType;
+    /// use std::path::Path;
+    ///
+    /// let rust = LanguageType::from_shebang(Path::new("./main.rs"));
+    ///
+    /// assert_eq!(rust, Some(LanguageType::Rust));
```

Comment:
```suggestion
    /// use std::path::Path;
    ///
    /// let rust = LanguageType::from_shebang("./main.rs");
    ///
    /// assert_eq!(rust, Some(LanguageType::Rust));
```

Reply:
That won't work as from_shebang expects a Path. Right now it directly takes from_path's argument `entry`. Should I change that?

Reply:
Just searched what AsRef<...> does, amazing! I'm loving your github workflow btw, congrats on that project.

Reply:
Thanks! I made it as a separate template if you'd like to use it for your own projects. https://github.com/XAMPPRocky/mean-bean-ci-template